evaluator/src/main/resources/rules/osadl.rules.kts

```diff
@@ -0,0 +1,94 @@
+import org.ossreviewtoolkit.evaluator.osadl.CompatibilityMatrix
+
+val ruleSet = ruleSet(ortResult, licenseInfoResolver) {
+    val licenseView = LicenseView.CONCLUDED_OR_DECLARED_AND_DETECTED
+
+    // Define a rule that is executed for each dependency.
+    dependencyRule("OSADL_MATRIX_COMPATIBILITY") {
+        // Requirements for the rule to trigger a violation.
+        require {
+            -isExcluded()
+        }
+
+        val projectLicenseInfo = licenseInfoResolver.resolveLicenseInfo(project.id).filter(licenseView)
+        val outboundLicenses = projectLicenseInfo.licenses.map { it.license }
+
+        // Define a rule that is executed for each license of the dependency.
+        licenseRule("OSADL_PROJECT_LICENSE_COMPATIBILITY", licenseView) {
+            outboundLicenses.forEach { outboundLicense ->
+                val compatibilityInfo = CompatibilityMatrix
+                    // Be conservative and use the simple license string without the exception string for lookup.
+                    // Eventually resulting false-positives are better than an incompatibility to be missed.
+                    .getCompatibilityInfo(outboundLicense.simpleLicense(), license.simpleLicense())
+
+                if (compatibilityInfo.compatibility in Compatibility.COMPATIBLE_VALUES) return@forEach
+
+                fun getSimplifiedLicenseString(license: SpdxSingleLicenseExpression) =
+                    if (license.exception() != null) {
+                        "${license.simpleLicense()} (simplified from '$license')"
+                    } else {
+                        "$license"
+                    }
+
+                val outboundLicenseString = getSimplifiedLicenseString(outboundLicense)
+                val inboundLicenseString = getSimplifiedLicenseString(license)
+
+                val projectCoords = project.id.toCoordinates()
+                val depCoords = dependency.id.toCoordinates()
+
+                when (compatibilityInfo.compatibility) {
+                    Compatibility.CONTEXTUAL -> warning(
+                        "Whether The outbound license $outboundLicenseString of project '$projectCoords' is " +
+                                "incompatible with the inbound license $inboundLicenseString of its dependency " +
+                                "'$depCoords' depends on the context. " + compatibilityInfo.explanation,
+                        "Get legal advice and eventually create a rule violation resolution."
+                    )
+
+                    Compatibility.UNKNOWN -> warning(
+                        "It is unknown whether the outbound license $outboundLicenseString of project " +
+                                "'$projectCoords' is incompatible with the inbound license $inboundLicenseString of " +
+                                "its dependency '$depCoords'. " + compatibilityInfo.explanation,
+                        "Get legal advice and eventually create a rule violation resolution."
```

Comment:
I believe using project specific rule violation resolutions is an unnecessarily expensive way to address this issue.
Ideally, the matrix would be updated but then it wouldn't be the OSADL matrix anymore. 

Can we do something to point the user in the right directing of trying to solve this globally?
Maybe it already helps a bit to say: "create a global rule violation resolution"?